### PR TITLE
Hide notices in wp-admin

### LINKED ISSUE #65: Hide notices that are a part of the setup task list
See #64.

A number of the extensions we are bundling show notices right after installation. We can suppress the ones we are including on the setup task list.

### PR DESCRIPTION
**Pull request by jeffikus**
Hides all plugin activation and wp-admin admin notices in the calypsoified wp-admin.

Closes #65

### CODE REVIEW

**Comment by justinshreve** [comment]
Hey @jeffikus - We recently added a linter to enforce WP php coding standards, and fixed a bunch of issues in the current files - so just a heads up there is some more rebasing to do here.

Is there anything @joshuatf and I can do to help clear up the remaining notifications?

**Comment by timmyc** [comment]
Hey @jeffikus wanted to check in and see if you still were going to work on this some more.

**Comment by jeffikus** [comment]
@timmyc yes, unless you have time? I'm going to tackle this tomorrow - I've been AFK sick most of this week, only got back today.

**Comment by timmyc** [comment]
Sorry you haven't been well this week @jeffikus - we are just departing our meetup today so I can't get to it, but can help out next week if still needed. I do have a layover today so I'll try to give your latest changes a test.

File: includes/helper-functions.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+	 * Constructor.
+	 */
+    private function __construct() {
+        // Silence
+    }
+
+    /**
+     * Remove Class Filter Without Access to Class Object
```
[the remark this answers is not in the document]
Comment (jeffikus) [reply includes/helper-functions.php]: Done in https://github.com/Automattic/wc-calypso-bridge/pull/92/commits/a879dac6d75f9f17ddd2f4273b39a143892abb5b

File: wc-calypso-bridge-class.php [hunk 2]
```diff
@@ -66,6 +71,45 @@ public function possibly_add_calypsoify_styles() {
...
+		foreach ( $extension_admin_notices_to_suppress as $class_name => $function_to_suppress ) {
+			WC_Calypso_Bridge_Helper_Functions::remove_class_action( 'admin_notices', $class_name, current( $function_to_suppress ), key( $function_to_suppress ) );
+		}
+		// List of extensions that do not use class level functions for admin notices.
+		$other_admin_notices = array( 'woocommerce_gateway_paypal_express_upgrade_notice', 'woocommerce_gateway_klarna_welcome_notice' );
+		foreach ( $other_admin_notices as $function_to_suppress ) {
+			remove_action( 'admin_notices', $function_to_suppress );
+		}
+		// Suppress: Looking for the store notice setting? It can now be found in the Customizer.
+		$updated = update_user_meta( get_current_user_id(), 'dismissed_store_notice_setting_moved_notice', true );
```
[the remark this answers is not in the document]
Comment (jeffikus) [reply wc-calypso-bridge-class.php]: Fixed in https://github.com/Automattic/wc-calypso-bridge/pull/92/commits/f6b3393ca3c1c48ddedd0190208755ec232e7d9c

**Comment by timmyc** [comment]
@justin do you have time to give this another review today?

File: class-wc-calypso-bridge-frontend.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'init', array( $this, 'remove_storefront_default_footer_credit' ), 10 );	
+    }
+    
+    /**
+	 * Unhooks Storefront footer credit
```
Comment (justinshreve) [inline class-wc-calypso-bridge-frontend.php]: Minor indentation/formatting issue here.

File: class-wc-calypso-bridge-frontend.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	/**
+	 * Display the new WordPress.com like theme credit
+	 *
+	 * @since  1.0.0
+	 * @return void
+	 */
+	public function wpcom_ecommerce_plan_storefront_credit() {
+		?>
+		<div class="site-info">
+			<?php echo esc_html( apply_filters( 'storefront_copyright_text', $content = '&copy; ' . get_bloginfo( 'name' ) . ' ' . date( 'Y' ) ) ); ?>
```
Comment (justinshreve) [inline class-wc-calypso-bridge-frontend.php]: $content isn't used again, and this line is a bit hard to follow. does

`echo esc_html( apply_filters( 'storefront_copyright_text', '&copy; ' . get_bloginfo( 'name' ) . ' ' . date( 'Y' ) ) );`

work instead?

File: class-wc-calypso-bridge-frontend.php [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+				<?php echo '<a href="https://wordpress.com/?ref=footer_website" target="_blank" title="' . esc_attr__( 'WordPress.com - The Best eCommerce Platform for WordPress', 'wc-calypso-bridge' ) . '" rel="author">' . esc_html__( 'Built with Storefront &amp; WordPress.com', 'wc-calypso-bridge' ) . '</a>.'; ?>
+			<?php } ?>
+		</div><!-- .site-info -->
+		<?php
+	}
+
+	/**
+	 * Loads required functionality, classes, and API endpoints.
+	 */
+	private function includes() {
```
Comment (justinshreve) [inline class-wc-calypso-bridge-frontend.php]: We don't need an `includes()` here.

File: class-wc-calypso-bridge-frontend.php [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+		}
+
+		return self::$instance;
+	}
+
+
+
+}
+
+WC_Calypso_Bridge_Frontend::instance();
```
Comment (justinshreve) [inline class-wc-calypso-bridge-frontend.php]: There should be a new line at the end of php files.

File: class-wc-calypso-bridge.php [hunk 7]
```diff
@@ -39,6 +39,49 @@ public function __construct() {
...
+		}	
+	}
+	
+	/**
+	 * 
+	 * Disables Specific Features within the Powerpack extension for Storefront.
+	 * 
+	 */
+	public function disable_powerpack_features() {
+		/**
```
Comment (justinshreve) [inline class-wc-calypso-bridge.php]: Minor indentation/formatting issue here.

File: class-wc-calypso-bridge.php [hunk 8]
```diff
@@ -39,6 +39,49 @@ public function __construct() {
...
+         * List of Powerpack features able to disable
```
Comment (justinshreve) [inline class-wc-calypso-bridge.php]: Should all of these be in the array below or does this comment need updated?

File: includes/class-wc-calypso-bridge-themes-setup.php [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+	}
+
+	/**
+	 * Constructor.
+	 */
+	private function __construct() {
+        $this->includes();
+
+        // Suppress WC Admin Notices
+		add_action( 'admin_head', array( $this, 'suppress_admin_notices' ) );
```
Comment (justinshreve) [inline includes/class-wc-calypso-bridge-themes-setup.php]: Indentation.

File: includes/class-wc-calypso-bridge-themes-setup.php [hunk 10]
```diff
@@ -0,0 +1,278 @@
+<?php
```
Comment (justinshreve) [inline includes/class-wc-calypso-bridge-themes-setup.php]: It looks like there is some formatting issues in this file that go against our PHP standards. Mostly indentation, but some lines with spacing issues like ` if (!$upload_file['error']) {`.

**Comment by jeffikus** [comment]
Refactored and separated out into #188
